ts/client/scripts/rebalancer.ts

```diff
@@ -0,0 +1,197 @@
+import { sendTransaction } from '../src/utils/rpc';
+
+// Env vars
+const CLUSTER: Cluster =
+  (process.env.CLUSTER_OVERRIDE as Cluster) || 'mainnet-beta';
+const CLUSTER_URL =
+  process.env.CLUSTER_URL_OVERRIDE || process.env.MB_CLUSTER_URL;
+const USER_KEYPAIR =
+  process.env.USER_KEYPAIR_OVERRIDE || process.env.MB_PAYER_KEYPAIR;
+const MANGO_ACCOUNT_PK = process.env.MANGO_ACCOUNT_PK || '';
+
+export interface OrderbookL2 {
+  bids: number[][];
+  asks: number[][];
+}
+
+async function rebalancer(): Promise<void> {
+  // Load client
+  const options = AnchorProvider.defaultOptions();
+  const connection = new Connection(CLUSTER_URL!, options);
+  const user = Keypair.fromSecretKey(
+    Buffer.from(
+      JSON.parse(
+        process.env.KEYPAIR || fs.readFileSync(USER_KEYPAIR!, 'utf-8'),
+      ),
+    ),
+  );
+  const userWallet = new Wallet(user);
+  const userProvider = new AnchorProvider(connection, userWallet, options);
+  const client = await MangoClient.connect(
+    userProvider,
+    CLUSTER,
+    MANGO_V4_ID[CLUSTER],
+    {
+      idsSource: 'get-program-accounts',
+    },
+  );
+
+  // Load mango account
+  let mangoAccount = await client.getMangoAccount(
+    new PublicKey(MANGO_ACCOUNT_PK),
+    true,
+  );
+  console.log(
+    `MangoAccount ${mangoAccount.publicKey} for user ${user.publicKey} ${
+      mangoAccount.isDelegate(client) ? 'via delegate ' + user.publicKey : ''
+    }`,
+  );
+  await mangoAccount.reload(client);
+
+  // Load group
+  const group = await client.getGroup(mangoAccount.group);
+  await group.reloadAll(client);
+  const usdcBank = group.getFirstBankByMint(
+    new PublicKey('EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v'),
+  );
+
+  // Loop indefinitely
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    await group.reloadAll(client);
+    mangoAccount = await mangoAccount.reload(client);
+    // console.log(mangoAccount.toString(group, true));
+
+    for (const tp of mangoAccount
+      .tokensActive()
+      .filter((tp) => tp.tokenIndex !== usdcBank.tokenIndex)) {
+      const baseBank = group.getFirstBankByTokenIndex(tp.tokenIndex);
+      const tokenBalance = tp.balanceUi(baseBank);
+
+      const serum3Markets = Array.from(
+        group.serum3MarketsMapByMarketIndex.values(),
+      )
+        // Find correct $TOKEN/$USDC market
+        .filter(
+          (serum3Market) =>
+            serum3Market.baseTokenIndex === tp.tokenIndex &&
+            serum3Market.quoteTokenIndex === usdcBank.tokenIndex,
+        );
+      if (!serum3Markets) {
+        continue;
+      }
+      const serum3Market = serum3Markets[0];
+      const serum3MarketExternal = group.serum3ExternalMarketsMap.get(
+        serum3Market.serumMarketExternal.toBase58(),
+      )!;
+      const maxBaseQuantity = serum3MarketExternal.baseSizeNumberToLots(
+        Math.abs(tokenBalance),
+      );
+      // Skip if quantity is too small
+      if (maxBaseQuantity.eq(new BN(0))) {
```

Comment:
too small means ==0? probably outdated comments

Reply:
no, basically if the quantity is too small for the lot size on the openbook market, at least this is what I am trying to achieve

Reply:
ah!